### PR TITLE
Part of #11159: ImageImporter::ImportResult uses std::vector

### PR DESCRIPTION
**Pull request by frutiemax**

### CODE REVIEW

**Comment by Gymnasiast** [comment]
I'd like a second opinion of @duncanspumpkin or @IntelOrca .

File: src/openrct2/CmdlineSprite.cpp [hunk 1]
```diff
@@ -246,7 +246,7 @@ static bool sprite_file_export(rct_g1_element* spriteHeader, const char* outPath
 
 static bool sprite_file_import(
```
Comment (duncanspumpkin) [inline src/openrct2/CmdlineSprite.cpp:247]: Can't we return a std::optional ImportResult? Then you can get rid of all the out parameters.

File: src/openrct2/CmdlineSprite.cpp [hunk 2]
```diff
@@ -266,17 +266,14 @@ static bool sprite_file_import(
...
-        *outBuffer = static_cast<uint8_t*>(result.Buffer);
-        *outBufferLength = static_cast<int>(result.BufferLength);
-        return true;
+        return importResult;
     }
     catch (const std::exception& e)
     {
         fprintf(stderr, "%s\n", e.what());
-        return false;
+        return importResult;
```
Comment (duncanspumpkin) [inline src/openrct2/CmdlineSprite.cpp:276]: Return a std::nullopt;

File: src/openrct2/CmdlineSprite.cpp [hunk 3]
```diff
@@ -244,10 +244,10 @@ static bool sprite_file_export(rct_g1_element* spriteHeader, const char* outPath
     }
 }
 
-static bool sprite_file_import(
-    const char* path, int16_t x_offset, int16_t y_offset, bool keep_palette, bool forceBmp, rct_g1_element* outElement,
-    uint8_t** outBuffer, int* outBufferLength, int32_t mode)
+static std::optional<ImageImporter::ImportResult> sprite_file_import(
+    const char* path, int16_t x_offset, int16_t y_offset, bool keep_palette, bool forceBmp, int32_t mode)
 {
+    ImageImporter::ImportResult importResult;
```
Comment (duncanspumpkin) [inline src/openrct2/CmdlineSprite.cpp:250]: No need to have this variable

File: src/openrct2/CmdlineSprite.cpp [hunk 4]
```diff
@@ -643,12 +638,8 @@ int32_t cmdline_for_sprite(const char** argv, int32_t argc)
...
         }
 
-        rct_g1_element spriteElement;
-        uint8_t* buffer;
-
-        int32_t bufferLength;
-        if (!sprite_file_import(
-                imagePath, x_offset, y_offset, false, false, &spriteElement, &buffer, &bufferLength, gSpriteMode))
+        auto importResult = sprite_file_import(imagePath, x_offset, y_offset, false, false, gSpriteMode);
+        if (!importResult)
```
Comment (tupaschoal) [inline src/openrct2/CmdlineSprite.cpp:642]:
```suggestion
        if (importResult == std::nullopt)
```
I'd do the `!importResult` too, but for consistency, the rest of the code usually does explicit comparison to `nullopt`

File: src/openrct2/CmdlineSprite.cpp [hunk 5]
```diff
@@ -768,14 +760,10 @@ int32_t cmdline_for_sprite(const char** argv, int32_t argc)
...
-            int bufferLength;
-
-            if (!sprite_file_import(
-                    imagePath.c_str(), x_offset == nullptr ? 0 : json_integer_value(x_offset),
-                    y_offset == nullptr ? 0 : json_integer_value(y_offset), keep_palette, forceBmp, &spriteElement, &buffer,
-                    &bufferLength, gSpriteMode))
+            auto importResult = sprite_file_import(
+                imagePath.c_str(), x_offset == nullptr ? 0 : json_integer_value(x_offset),
+                y_offset == nullptr ? 0 : json_integer_value(y_offset), keep_palette, forceBmp, gSpriteMode);
+            if (!importResult)
```
Comment (tupaschoal) [inline src/openrct2/CmdlineSprite.cpp:766]:
```suggestion
            if (!importResult  == std::nullopt)
```

**Pull request merged**